server/src/main/java/com/cloud/server/ManagementServerImpl.java

```diff
@@ -1326,10 +1327,9 @@ private HypervisorType getHypervisorType(VMInstanceVO vm, StoragePool srcVolumeP
             if (s_logger.isDebugEnabled()) {
                 s_logger.debug("Searching for all hosts in cluster " + cluster + " for migrating VM " + vm);
             }
-            allHostsPair = searchForServers(startIndex, pageSize, null, hostType, null, null, null, cluster, null, keyword, null, null, null, null);
-            // Filter out the current host.
+            allHostsPair = searchForServers(startIndex, pageSize, null, hostType, null, null, null, cluster, null, keyword, null, null, null,
+                null, srcHost.getId());
             allHosts = allHostsPair.first();
-            allHosts.remove(srcHost);
             plan = new DataCenterDeployment(srcHost.getDataCenterId(), srcHost.getPodId(), srcHost.getClusterId(), null, null, null);
         }
 
```

Comment:
@davidjumani can you check if `otherHosts` here is not required as there is no change in the `allHosts` & its count, which you can return as it is (as a List), instead of Pair (with count). You can use use the `allHosts` count wherever applicable.

Reply:
It's needed here since the data type is different `Pair<List<HostVO>, Integer>` vs `Pair<List<? extends Host>, Integer>`

Reply:
thanks for confirming that @davidjumani